### ISSUE TITLE
cadvisor takes quite high %cpu

### ISSUE THREAD

**Issue by hans-d** [kontena/kontena#395]
seeing almost 15% + cpu usage for cadvisor, with much higher spikes (40%+) - as seen with top
Currently running 13 containers on that node, being a 4 core running Ubuntu

**Comment by jakolehm** [comment]
Is docker using aufs?

**Comment by hans-d** [comment]
this one: yes

**Comment by jakolehm** [comment]
I'm pretty sure that cadvisor tries to track disk usage and that's consuming cpu cycles. cAdvisor disk usage tracking is only supported when docker uses aufs, although latest versions do track also overlayfs. High cpu usage is the reason why we haven't yet upgraded to latest cadvisor version.

**Comment by hans-d** [comment]
seeing now 100% +. Is there a way to limit cadvisor in some way, or even disable it?

**Comment by jakolehm** [comment]
You can only stop it manually by issuing `docker stop kontena-cadvisor` .. of course it will come back after kontena-agent starts. But you can bring your own cadvisor image to node if you want, just set `CADVISOR_IMAGE` and `CADVISOR_VERSION` env variables to agent.

Btw do you have high number of containers when this happens?

**Comment by hans-d** [comment]
I've indeed stopped in manually for a while.
Part of the CPU% usage by cadvisor was also caused by a continuously restarting LB. Now I've fixed that, i've started cadvisor again and so far it seems to behave.
This issue on the node also seemed to affect the vps where the master was running on. I could hardly connect to the VPS. After killing both the cadvisor container and the reloading LB stuff settled down.
